federation/pkg/federation-controller/secret/secret_controller.go

```diff
@@ -55,34 +55,37 @@ var (
 	RequiredResources = []schema.GroupVersionResource{apiv1.SchemeGroupVersion.WithResource("secrets")}
 )
 
-type SecretController struct {
-	// For triggering single secret reconciliation. This is used when there is an
-	// add/update/delete operation on a secret in either federated API server or
-	// in some member of the federation.
-	secretDeliverer *util.DelayingDeliverer
-
-	// For triggering all secrets reconciliation. This is used when
-	// a new cluster becomes available.
+// FederationSyncController synchronizes the state of a federated type
+// to clusters that are members of the federation.
+type FederationSyncController struct {
```

Comment:
Yes, this is just the start of the work.  There was initially concern that the approach would require an extensive design, and this PR was an attempt to make the intended approach clear.

This is a strict refactor - there are no functional changes - so I'm not sure @mwielgus's attention is required.  I hear he has his hands full with non-federation responsibilities.